test/testpreprocessor.cpp

```diff
@@ -246,6 +246,8 @@ class TestPreprocessor : public TestFixture {
         TEST_CASE(testDirectiveIncludeComments);
 
         TEST_CASE(testSameLine);  // #7912
+
+        TEST_CASE(macro_argif);
```

Comment:
I don't see how this `macro_argif` testcase is related to undefines.

Reply:
Yes, it should be part of other PR.